src/radical/pilot/pmgr/launching/default.py

```diff
@@ -841,7 +694,7 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand, tar_name):
         rp_version              = rcfg.get('rp_version')
         virtenv_mode            = rcfg.get('virtenv_mode',        DEFAULT_VIRTENV_MODE)
         virtenv                 = rcfg.get('virtenv',             default_virtenv)
-        cores_per_node          = rcfg.get('cores_per_node', 0)
+        cores_per_node          = rcfg.get('cores_per_node', 1)
```

Comment:
for some systems this parameter is not set in a resource config, and for such cases default value `1` will mess up with `number_cores` and in this case it is tricky to use for `number_nodes` below.